src/lib/analytics/apis.web.js

```diff
@@ -51,15 +48,25 @@ class FullStoryWrapper {
 }
 
 class GoogleWrapper {
+  dataLayer = null
+
+  constructor(dataLayer) {
+    assign(this, { dataLayer })
+  }
+
   logEvent(event: string, data: any = {}) {
-    dataLayer.push({ event, ...data })
+    this.dataLayer.push({ event, ...data })
   }
 }
 
-export default pickBy({
-  mautic: mt,
-  sentry: SentryWeb,
-  googleAnalytics: dataLayer ? new GoogleWrapper() : null,
-  amplitude: amplitude.getInstance(),
-  fullStory: FS ? new FullStoryWrapper() : null,
-})
+export default () => {
+  const { mt, FS, dataLayer } = window
+
+  return pickBy({
+    mautic: mt,
+    sentry: SentryWeb,
+    googleAnalytics: dataLayer ? new GoogleWrapper(dataLayer) : null,
+    amplitude: amplitude.getInstance(),
+    fullStory: FS ? new FullStoryWrapper(FS, window) : null,
```

Comment:
why do you suddenly need to pass dataLayer, FS, window to the constructors... its redudant, the way it was before is fine

Reply:
we couldn't do as before because we couldn't destructurize window to mt FS and data layer globally in this module.
so we're getting inside default-exported function which will be lazy called on initAnalytics, then passing refs to the constructors
it's not redudant, it's an DI pattern commonly used